tsl/test/sql/compression_update_delete.sql

```diff
@@ -1446,3 +1446,37 @@ DELETE FROM test_limit WHERE id > 0;
 \set ON_ERROR_STOP 1
 
 DROP TABLE test_limit;
+
+-- check partial compression with DML
+CREATE TABLE test_partials (time timestamptz NOT NULL, a int, b int);
+SELECT create_hypertable('test_partials', 'time');
+INSERT INTO test_partials
+VALUES -- chunk1
+  ('2020-01-01 00:00'::timestamptz, 1, 2),
+  ('2020-01-01 00:01'::timestamptz, 2, 2),
+  ('2020-01-01 00:04'::timestamptz, 1, 2),
+  -- chunk2
+  ('2021-01-01 00:00'::timestamptz, 1, 2),
+  ('2021-01-01 00:04'::timestamptz, 1, 2),
+  -- chunk3
+  ('2022-01-01 00:00'::timestamptz, 1, 2),
+  ('2022-01-01 00:04'::timestamptz, 1, 2);
+-- enable compression, compress all chunks
+ALTER TABLE test_partials SET (timescaledb.compress);
+SELECT compress_chunk(show_chunks('test_partials'));
+-- fully compressed
+EXPLAIN (costs off) SELECT * FROM test_partials ORDER BY time;
+-- verify correct results
+SELECT * FROM test_partials ORDER BY time;
+-- check that DML causes transparent decompression and that
+-- data gets shifted to the uncompressed parts
+EXPLAIN (costs off) DELETE FROM test_partials WHERE time <> ALL(SELECT time from test_partials);
```

Comment:
This delete query shouldn't actually touch anything, right? So the fact that the data is decompressed here is a missing optimization that we should add at some point. Can you change this to actually delete one row from a compressed batch, to make the test more robust? Also might be good to add one chunk with at least two compressed batches, so that both parts of the partial chunk path are tested.

Reply:
@akuzm I believe it could be a `chicken and egg` thingy because unless we decompress we might not get access to all the values that we need to check against.

Reply:
@akuzm there are other tests which test out various scenarios with a mix of multiple partial, complete compressed, uncompressed chunks in tsl/test/sql/compression_ddl.sql already.

Reply:
> @akuzm I believe it could be a `chicken and egg` thingy because unless we decompress we might not get access to all the values that we need to check against.

I mean, imagine in the future we have a more optimal path for DELETEs, for example through TAM, and it's not going to decompress anything if the WHERE doesn't match anything. So the test will stop working. It would be good if some rows actually matched the clause and were deleted, e.g. if it was `time >= ALL(...`, then the latest row would match.

Reply:
Ok, changed